website/project/issue-pr-grading.rst

```diff
@@ -0,0 +1,418 @@
+   closing this issue, but will reopen if we decide to revisit this”
+-  “The `Circle` class has been implemented, except for the intersection
+   method, which turned out to be more complex than anticipated. I am
+   closing this issue, and have created a separate task (#5) for
+   implementing circle intersections”
+
+If multiple developers are assigned to the issue, the closing statement
+must also include a clear statement of what work was done by each
+developer
+
+Requirements for Pull Requests
+------------------------------
+
+As in many software projects, the ``main`` branch of our Git
+repository should only contain stable code that compiles, runs, and
+passes all tests. This is enforced by making the main branch a
+*protected* branch, meaning that only the instructors and TAs have the
+ability to directly push to that branch.
+
+On top of that, there is an additional ``dev`` (short for “development”)
+branch that acts as a “pre-main” branch. The ``dev`` branch is also
+protected, and should only contain stable code, but it is used as a
+staging area to integrate code (and uncover potential integration
+issues) before publishing code to ``main``.
+
+This means that, to get code into ChiGame, you must make pull request
+to get code merged into the ``dev`` branch (and the instructor and TAs
+will decide when to merge that code into ``main``).
+
+Because the ``dev`` branch should contain stable code, pull requests to
+``dev`` must be carefully reviewed before they are approved. If you want
+to merge code into ``dev``, your code must be reviewed by (1) someone in
+your team and (2) your senior developer. Please note that the senior
+developer will only do their code review once your teammate has completed
+their review. Additionally, when your code affects code
+owned by another team, you must have an additional reviewer from each
+team whose code is affected.
+
+A pull request should contain a relatively self-contained piece of work.
+In fact, it is not uncommon for a task (or, at most, 2-3 tasks) to
+result in a pull request. We encourage you to touch base with your
+senior developer before creating your first pull requests, to determine
+whether the work you want to merge merits a pull request (or, similarly,
+whether it needs to be broken up into multiple pull requests).
+
+Like a task, a pull request should have a descriptive title, a
+descriptive summary, updates on the progress of the pull request
+(especially since changes may need to be done based on the code
+reviews), and a closing statement. The assignees for the pull request
+must be everyone who contributed to the code that is being merged.
+
+Before a pull request is approved, the merge being merged must be
+"up to date" with ``dev``, meaning that any changes that happened
+in ``dev`` while the pull request was being reviewed must be merged
+into the pull request's branch. It is sometimes not uncommon for
```

Comment:
Is there any type of etiquette we want to include for making branches / naming them?